### ISSUE TITLE
Exception notices are now only displayed in English

### ISSUE THREAD

**Issue by xchs** [contao/contao#453]
**Affected version(s)**

Contao 4.7.4

**Description**

In Contao 4.7.4, the exception notices are only displayed in English, and not according to the browser language.

**Contao 4.7.4:**

![01](https://user-images.githubusercontent.com/1218766/56691105-603f1d80-66df-11e9-99c9-1ada39842240.jpg)

**Contao 4.4.38:**

![02](https://user-images.githubusercontent.com/1218766/56691134-7816a180-66df-11e9-98e7-16e157700a1a.jpg)

**How to reproduce** 

* Set up a fresh Contao 4.7 installation (without configuring anything in the Contao back end)
* Open the front end so that you get the "_No root page found_" exception

**Comment by leofeyer** [comment]
Does this affect the "no root page found" exception only? Because we have added a special routine for this case in Contao 4.7, which might lack proper language handling:

https://github.com/contao/contao/blob/388edab7cad4d8bdd0218bd2586d32847a3beea9/core-bundle/src/EventListener/ExceptionConverterListener.php#L58-L60

**Comment by xchs** [comment]
> Does this affect the "no root page found" exception only?

No. I also get an English notice if, for example, I just create a root page and **no** regular pages.

**Comment by leofeyer** [comment]
@aschempp It seems that the `RouterListener` has priority 32, whereas the `TranslatorListener` has priority 10. 🙄 That is why it works for e.g. the `NoLayoutSpecifiedException` but not for any exceptions thrown during routing.

**Comment by leofeyer** [comment]
I have tested all exceptions that the `PrettyErrorScreenListener` class is supposed to handle:

| Class | Status | Behavior |
| --- | --- | --- |
| `ForwardPageNotFoundException` | **WRONG** | Shows the default "Oops! An Error Occured" page instead of a pretty error screen |
| `IncompleteInstallationException` | **WRONG** | Is ignored completely in the front end in production (WTF?) |
| `InsecureInstallationException` | **WRONG** | Outputs the raw PHP error message instead of a pretty error screen (you need to comment out the `@ini_set('display_errors', '0');` line in the `app.php` entry point) |
| `InvalidRequestTokenException` | **WRONG** | Renders the pretty error screen but shows the text in English even though the root language is German |
| `NoActivePageFoundException` | **WRONG** |Renders the pretty error screen but shows the text in English even though the root language is German |
| `NoLayoutSpecifiedException` | OK |Behaves correctly (pretty error screen in German) |
| `NoRootPageFoundException` | OK | Renders the pretty error screen but shows the text in English, which is kind of correct as there is no root page with a different language |

**Comment by xchs** [comment]
> Renders the pretty error screen but shows the text in English, which is kind of correct as there is no root page with a different language

Yes, but with Contao 4.4 the pretty error screen is rendered in German (under the same circumstances).

**Comment by leofeyer** [comment]
What is interesting about the "insecure installation" problem is that the final exception is the "resource not found" exception from the routing component:

```
Uncaught Symfony\Component\Routing\Exception\ResourceNotFoundException: None of the routers in the chain matched this request HEAD /app.php/ HTTP/1.1 Accept: application/vnd.t42.header-replay 
```

**Comment by leofeyer** [comment]
Another thing we need to fix: If there are no pages at all and the "insecure installation" exception is thrown in the front end, the final exception will be the "resource not found" exception from the routing, because there is no root for `/`:

```
Uncaught Symfony\Component\Routing\Exception\ResourceNotFoundException: None of the routers in the chain matched this request HEAD /app.php/ HTTP/1.1 Accept: application/vnd.t42.header-replay 
```

**Comment by leofeyer** [comment]
The "insecure installation" problem is caused by ec82a9745ad8783799ec563878f4676b24be611e by the way, because the accept header is `application/vnd.t42.header-replay` in this case. It could be fixed by adding the following to the `InsecureInstallationListener` class:

```php
// TODO: remove this once we are getting rid of header replaying
if (\in_array(HeaderReplayListener::CONTENT_TYPE, $request->getAcceptableContentTypes(), true)) {
    return;
}
```

Although then Contao behaves like this:

| Path | Status | Behavior |
| --- | --- | --- |
| `/` | **NOK** | Renders the pretty error screen in English, although the root page is in German |
| `/contao/install` | OK | Renders the pretty error screen in German |
| `/contao/login` | **NOK** | Renders the general "internal server error" screen instead of the "insecure installation" error screen |

**Comment by xchs** [comment]
Huh, it seems to be more complex than I actually thought. :see_no_evil:

**Comment by leofeyer** [comment]
@aschempp Could this be related to #456?

**Comment by leofeyer** [comment]
Fixed in 2a33cd4e7bb8e6bb9fc2dbb4b81d6124547468b4. I have opened a separate issue regarding the "no root page found" exception: #593
